packages/ckeditor5-engine/src/view/matcher.js

```diff
@@ -316,31 +430,37 @@ function matchClasses( patterns, element ) {
 // @param {module:engine/view/element~Element} element Element which styles will be tested.
 // @returns {Array|null} Returns array with matched style names or `null` if no styles were matched.
 function matchStyles( patterns, element ) {
-	const match = [];
+	const styles = [];
 
-	for ( const name in patterns ) {
-		const pattern = patterns[ name ];
+	element.getStyleNames()
+		.forEach( style => {
+			// Add the direct style (might be a shorthand).
+			styles.push( [ style, element.getStyle( style ) ] );
```

Comment:
It was actually surprising for me that we don't have a method of returning styles similar to `element.getAttributes()`. But I see that actually `getStyleNames` is using private `_getStyleEntries` method:
https://github.com/ckeditor/ckeditor5/blob/a8b34f05482d25ac6ecbfa60b073e9799c6412e3/packages/ckeditor5-engine/src/view/stylesmap.js#L357-L365
Maybe we should add `element.getStyles()` method instead?